core/src/main/java/com/google/cloud/sql/core/CredentialFactoryProvider.java

```diff
@@ -45,4 +46,40 @@ static CredentialFactory getCredentialFactory() {
     }
     return credentialFactory;
   }
+
+  static CredentialFactory getInstanceCredentialFactory(
+      CredentialFactory defaultCredentialFactory, ConnectorConfig config) {
```

Comment:
You are right that this is weird. I'll rework CredentialFactoryProvider a bit so that this makes more sense. We want to reuse the defaultCredentialFactory instance.